### ISSUE TITLE
click.echo does not force colors on Windows with color=True

### ISSUE THREAD

**Issue by krisztianloki** [pallets/click#2606]
Using `click.echo(click.style("Foo", fg="red"), color=True)` on Windows will not print "Foo" in red if the output is not a terminal that supports colors, so it seems that `color=True` is ignored on Windows.

Here is an example code to reproduce the bug:
```
import click


def main():
    test_txt = "click_echo.txt"
    with open(test_txt, "w") as f:
        click.secho("Without setting color", file=f, fg="red")
        click.secho("With color=False", file=f, fg="red", color=False)
        click.secho("With color=True", file=f, fg="red", color=True)

    click.echo(f"Check the contents of {test_txt}")


if __name__ == "__main__":
    main()
```
After running it on Linux the third line of `click_echo.txt` will contain the ANSI color codes but on Windows all three lines will be the same (the behavior is the same if `echo` prints on the console and the script output is redirected to a file)

Environment:

- Python version: 3.9
- Click version: 8.1.7

**Comment by davidism** [comment]
Windows uses a different mechanism for color than Unix. It's not possible to force it on. Or if it is, you'd need to take that up with colorama.

**Comment by krisztianloki** [comment]
I've looked into it and it seems that there is a bug in `click` itself: the `color` parameter is not passed when `auto_wrap_for_ansi()` is called on line 317 in `utils.py`:
```
    # ANSI style code support. For no message or bytes, nothing happens.
    # When outputting to a file instead of a terminal, strip codes.
    else:
        color = resolve_color_default(color)

        if should_strip_ansi(file, color):
            out = strip_ansi(out)
        elif WIN:
            if auto_wrap_for_ansi is not None:
                file = auto_wrap_for_ansi(file)  # type: ignore
            elif not color:
                out = strip_ansi(out)
```
If I change it to `file = auto_wrap_for_ansi(file, color)` then forcing colors seems to work.

**Comment by davidism** [comment]
Thanks for looking. Happy to review a PR.

**Comment by Tyl13** [comment]
Files are not ttys and therefore will not print color. We only write color to terminals. Have a nice day.

**Comment by kdeldycke** [comment]
FWIW, I had to skip color-based tests on Windows in a lot of my Click-based libraries:
- https://github.com/kdeldycke/meta-package-manager/commit/5ecbe5b4557fa338c53e42061d588cd62e7f8067
- https://github.com/kdeldycke/click-extra/commit/43419461ca732869ddf0e37214a41b957dcf7de5

So yes, I am pretty sure there is something wrong with colors on Windows. But I never found a way to reliably fix it and find the root cause.

Even with loads of hacks and workarounds in my [Click Extra](https://github.com/kdeldycke/click-extra) project, like:
- https://github.com/kdeldycke/click-extra/blob/44732144e3f1f4a3da059a084e02c4c3c8f95579/click_extra/commands.py#L96-L124
- https://github.com/kdeldycke/click-extra/blob/44732144e3f1f4a3da059a084e02c4c3c8f95579/click_extra/colorize.py#L258-L269

So yeah, @Tyl13 is probably onto something here with his #2731 PR ! :)

Related issues might include:
- https://github.com/pallets/click/issues/2110
- https://github.com/pallets/click/issues/2111

**Comment by AndreasBackx** [comment]
Fixed by #2607
